### PR TITLE
[App Service] `az functionapp create`, `az functionapp list-flexconsumption-runtimes`: Add support of Stacks API for Flex SKUs

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**
- az functionapp create
- az functionapp deployment github-actions
- az functionapp config appsettings set
- az functionapp list-flexconsumption-runtimes
- az functionapp runtime config set

**Description**
Adding the stacks API support for Flex SKU. 

**Testing Guide**
_Create a flex function app_

`az functionapp create -g flex-ea -n <function app name> -s <storage account name> --functions-version 4 --debug --runtime python --runtime-version 3.11 --flexconsumption-location '"eastasia"' --debug --deployment-storage-auth-type storageAccountConnectionString`

_List  flex stacks_

`az functionapp list-flexconsumption-runtimes --location eastasia --runtime python`

**History Notes**
[AppService] `az functionapp create`: Will validate that the provided runtime is supported by flex when creating a flex function app
[AppService] `az functionapp list-flexconsumption-runtimes`: Add support for this new command so that it provides the list of supported flex runtimes when provided the `--location` and `--runtime`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Review by arroyc — APPROVED** [review]

**Review by amamounelsayed — APPROVED** [review]

**Comment by zhoxing-ms** [comment]
> [AppService] az functionapp create: Create _FlexFunctionAppStackRuntimeHelper for the stacks API call needed for flex function app creation
[AppService] az functionapp list-flexconsumption-runtimes: Use _FlexFunctionAppStackRuntimeHelper when viewing list of supported flex runtimes
[AppService] az functionapp runtime config set: Use _FlexFunctionAppStackRuntimeHelper to check if provided runtime is supported by flex
[Appservice] az functionapp config appsettings set: Use _FlexFunctionAppStackRuntimeHelper to check if the runtime has reached its end of life date

Actually, these `History Notes` are customer-oriented. Please write a functional description of the changes in the customer interface, such as adding or modifying commands or parameters, etc

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -187,6 +187,10 @@ def load_arguments(self, _):
     with self.argument_context('functionapp list-runtimes') as c:
         c.argument('os_type', options_list=["--os", "--os-type"], help="limit the output to just windows or linux runtimes", arg_type=get_enum_type([LINUX_OS_NAME, WINDOWS_OS_NAME]))
 
+    with self.argument_context('functionapp list-flexconsumption-runtimes') as c:
+        c.argument('location', help="limit the output to just the runtimes available in the specified location")
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:191]:
```suggestion
        c.argument('location', arg_type=get_location_type(self.cli_ctx), help="limit the output to just the runtimes available in the specified location")
```

**Comment by kamperiadis** [comment]
> > [AppService] az functionapp create: Create _FlexFunctionAppStackRuntimeHelper for the stacks API call needed for flex function app creation
> > [AppService] az functionapp list-flexconsumption-runtimes: Use _FlexFunctionAppStackRuntimeHelper when viewing list of supported flex runtimes
> > [AppService] az functionapp runtime config set: Use _FlexFunctionAppStackRuntimeHelper to check if provided runtime is supported by flex
> > [Appservice] az functionapp config appsettings set: Use _FlexFunctionAppStackRuntimeHelper to check if the runtime has reached its end of life date
> 
> Actually, these `History Notes` are customer-oriented. Please write a functional description of the changes in the customer interface, such as adding or modifying commands or parameters, etc

Thank you @zhoxing-ms I have corrected the History Notes.

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py [hunk 2]
```diff
@@ -475,24 +474,26 @@ def test_functionapp_workloadprofiles(self, resource_group, storage_account):
...
 
-        self.cmd('containerapp env create --name {} --resource-group {} --location {} --enable-workload-profiles  --logs-destination none'.format(
+        self.cmd('containerapp env create --name {} --resource-group {} --location northeurope --enable-workload-profiles  --logs-destination none'.format(
             managed_environment_name,
             resource_group,
-            location,
         ))
-        
+
+        time.sleep(260)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:482]: Why do we need such a long sleep time? In addition, do we only need to re-record these tests in live mode?
```suggestion
        if self.is_live:
              time.sleep(260)
```

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py [hunk 3]
```diff
@@ -517,6 +518,7 @@ def test_functionapp_workloadprofiles(self, resource_group, storage_account):
             workload_profile_name_2
         ))
 
+        time.sleep(1200)
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:521]: Same as above

File: src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py (hunk 2 above)
Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_functionapp_commands.py:482]: Sometimes it takes longer for the resources to be created so if we try to run commands with resources that have not yet finished getting created, we get errors. The Centauri workload profiles and dapr tests specially take a longer time to run that other tests.

**Review by amamounelsayed — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**